### PR TITLE
[141][144] Implement accept/reject job offer API @PUT

### PR DESCRIPTION
**Pull request by badnameok**
## What did you do

- Implemented API endpoint to allow actor to accept job offers sent by the casting

## Demo

- login as actor on swagger
- create a job application
- login as casting of the respective job or use prisma studio to change `ApplicationStatus` to `OFFER_SENT`

### Testing Success
- use the accept/reject API with id of the respective `JOB` (not `APPLICATION`!)
- see in prisma studio if it is successful
### Testing Failure
- try using the same API again, it should become a bad request.
- try using arbitrary `jobId`, it should be a "haven't applied to this job" or "not found"

### CODE REVIEW

File: apps/api/src/modules/job/application/offer/offer.service.ts [hunk 1]
```diff
@@ -38,4 +38,48 @@ export class OfferService {
...
+      await this.applicationRepository.getApplicationbyActorJob(actorId, jobId)
+    if (!application) {
+      throw new BadRequestException('You have not applied to this job')
+    }
+    if (application.status === ApplicationStatus.OFFER_ACCEPTED) {
+      throw new BadRequestException('You have already accepted this job offer')
+    }
+    if (application.status !== ApplicationStatus.OFFER_SENT) {
+      throw new BadRequestException('You cannot accept this job offer')
+    }
```
Comment (palmcm) [inline apps/api/src/modules/job/application/offer/offer.service.ts:57]: I think we don't need to check for specific status
If so you might done for other offer status
wait for @miello @PKhing to confirm

**Review by PKhing — COMMENTED** [review]
How can I test if you don't merge it into dev

**Comment by badnameok** [comment]
> How can I test if you don't merge it into dev

pushed

File: apps/api/src/modules/job/application/offer/offer.service.ts [hunk 2]
```diff
@@ -36,6 +69,32 @@ export class OfferService {
...
+      userId: actorId,
+      jobId: job.jobId,
+      actorId: actorId,
+      type: NotificationType.RECEIVE_OFFER,
+    })
     return { message: 'Job offer sent' }
   }
+
+  async acceptJobOffer(jobId: number, actorId: number) {
+    this.updateApplication(
```
Comment (miello) [inline apps/api/src/modules/job/application/offer/offer.service.ts:82]:
```suggestion
    await this.updateApplication(
```

File: apps/api/src/modules/job/application/offer/offer.service.ts [hunk 3]
```diff
@@ -36,6 +69,32 @@ export class OfferService {
...
+      jobId,
+      actorId,
+      'accept',
+      ApplicationStatus.OFFER_ACCEPTED,
+      NotificationType.ACCEPT_OFFER,
+    )
+  }
+
+  async rejectJobOffer(jobId: number, actorId: number) {
+    this.updateApplication(
```
Comment (miello) [inline apps/api/src/modules/job/application/offer/offer.service.ts:92]:
```suggestion
    await this.updateApplication(
```

**Review by miello — CHANGES_REQUESTED** [review]

**Review by miello — APPROVED** [review]

**Pull request merged**